documentation/DPG-quickstart.md

```diff
@@ -0,0 +1,298 @@
+
+   ```yaml
+   parameters:
+     "service-dir":
+       default: "YOUR_SERVICE_DIRECTORY"
+
+   emit: ["@azure-tools/typespec-ts"]
+
+   options:
+     "@azure-tools/typespec-ts":
+       package-dir: "YOUR_SERVICE_FOLDER"
+       isModularLibrary: true
+       packageDetails:
+         name: YOUR_PACKAGE_NAME
+         description: "SHORT_DESCRIPTION"
+         version: "1.0.0-beta.1"
+   ```
+
+2. **Configure tsp-location.yaml in SDK repository**  
+   
+   You can update `tsp-location.yaml` under project root folder to set the typespec project. You can refer to the [tsp-location.yaml](https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/TypeSpec-Project-Scripts.md#tsp-locationyaml) which describes the supported properties in the file.
+   
+   If this is the first time to generate code you need to create the project folder.
+   ```shell
+   mkdir -p sdk/agrifood/agrifood-farming # create project folder
+   cd sdk/agrifood/agrifood-farming # enter project folder
+   ```
+   Then prepare the `tsp-location.yaml` and here is an example:
+
+   ```yaml
+   directory: specification/agrifood/DataPlane
+   commit: b646a42aa3b7a0ce488d05f1724827ea41d12cf1 # the commit id you'd like to refer for generation
+   repo: Azure/azure-rest-api-specs
+   ```
+
+3. **Generate code**
+
+    Run the following two scripts from project directory (i.e sdk/agrifood/agrifood-farming) to generate the code:
+
+    > NOTE: These scripts require PowerShell version 7 or higher
+    ```shell
+    pwsh ../../../eng/common/scripts/TypeSpec-Project-Sync.ps1 .
+    ```
+
+    followed by
+
+    ```shell
+    pwsh ../../../eng/common/scripts/TypeSpec-Project-Generate.ps1 .
+    ```
+
+    The version of typespec-ts is configured in [emitter-package.json](https://github.com/Azure/azure-sdk-for-js/blob/main/eng/emitter-package.json) and relevant lock file [emitter-package-lock.json](https://github.com/Azure/azure-sdk-for-js/blob/main/eng/emitter-package-lock.json). Change them in local, if you would like to use a different version of typespec-ts.
+
+4. **Edit rush.json**  
+    
+    As the libraries in azure-sdk-for-js repository are managed by rush, you need to add an entry in rush.json under projects section for the first time to make sure it works. For example:
+
+    ```
+        {
+          "packageName": "@azure/agrifood-farming",
+          "projectFolder": "sdk/agrifood/agrifood-farming",
+          "versionPolicyName": "client"
+        },
+    ```
+
+    Here you also need to replace the `packageName`, `projectFolder` into your own services'.
+
+5. **Build your project**  
+
+    After this finishes, you will see the generated code in `src` folder in your **{PROJECT_ROOT}**.  
+    To get a workable package, you can run the following commands to get a artifact if you like.
+
+    ```shell
+    rush update
```

Comment:
lets mention that Rush has to be installed `npm install @microsoft/rush -g`

@deyaaeldeen pointed out in my previous PR that I forgot to add `-g` lets make sure we have it